addons/account/migrations/11.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 bloopark systems (<http://bloopark.de>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+# backup of state column with old / non existing values 'proforma' and
+# 'proforma2'
+column_copies = {
+    'account_invoice ': [
+        ('state', None, None),
+    ],
+}
+
+
+def add_required_colum_chart_template_id(cr):
+    """The model account.reconcile.model.template has now a new required
+    many2one field chart_template_id with no default value. To avoid errors
+    on module update we add already the column here and fill it with the
+    right chart template ID."""
```

Comment:
It should not be a fatal error if a uniqueness constraint cannot be set on a column during a module upgrade. I believe the chart template id is set by each of the localization's data files, so you can leave out this whole bit if I am not mistaken.

Reply:
Interesting. I didn't know that it would work but I tested it and yes it works. You will get first a warning in your server log but later the chart_template_id field will be set by the XML record (l10n_fr has a chart_template_id for example).